projects/rocprofiler-sdk/tests/pytest-packages/tests/rocprofv3.py

```diff
@@ -231,3 +231,192 @@ def test_rocpd_data(
+    }
+
+    keys_mapping = {
+        "kernel": {
+            "Thread_Id": ("thread_id", None),
+            "Correlation_Id": ("correlation_id", "internal"),
+            "Start_Timestamp": ("start_timestamp", None),
+            "End_Timestamp": ("end_timestamp", None),
+            "Queue_Id": ("dispatch_info.queue_id.handle", None),
+            "Kernel_Id": ("dispatch_info.kernel_id", None),
+            "Dispatch_Id": ("dispatch_info.dispatch_id", None),
+            "Stream_Id": ("stream_id.handle", None),
+            "Workgroup_Size_X": ("dispatch_info.workgroup_size.x", None),
+            "Workgroup_Size_Y": ("dispatch_info.workgroup_size.y", None),
+            "Workgroup_Size_Z": ("dispatch_info.workgroup_size.z", None),
+            "Grid_Size_X": ("dispatch_info.grid_size.x", None),
+            "Grid_Size_Y": ("dispatch_info.grid_size.y", None),
+            "Grid_Size_Z": ("dispatch_info.grid_size.z", None),
+        },
+        "memory_allocation": {
+            "Operation": (),  # Special case
+            "Correlation_Id": ("correlation_id", "internal"),
+            "Start_Timestamp": ("start_timestamp", None),
+            "End_Timestamp": ("end_timestamp", None),
+        },
+        "memory_copy": {
+            "Correlation_Id": ("correlation_id", "internal"),
+            "Start_Timestamp": ("start_timestamp", None),
+            "End_Timestamp": ("end_timestamp", None),
+        },
+        "regions": {
+            "Thread_Id": ("thread_id", None),
+            "Correlation_Id": ("correlation_id", "internal"),
+            "Start_Timestamp": ("start_timestamp", None),
+            "End_Timestamp": ("end_timestamp", None),
+        },
+    }
+
+    for data in csv_data:
+        filename, _csv_data = data
+        file_category = [category for category in categories if category in filename]
+        assert len(file_category) > 0, f"{filename} is not a valid csv filename"
+        category = file_category[0]
+        if category == "counter_collection":
+            _js_data = json_data["rocprofiler-sdk-tool"]["callback_records"][category]
+        elif category == "agent":
+            _js_data = json_data["rocprofiler-sdk-tool"]["agents"]
+        elif category == "regions":
+            buffer_records = json_data["rocprofiler-sdk-tool"]["buffer_records"]
+            _js_data = []
+            for key, value in buffer_records.items():
+                if key == "marker_api":
+                    string_records = []
+                    marker_records = json_data["rocprofiler-sdk-tool"]["buffer_records"][
+                        "marker_api"
+                    ]
+                    for item in json_data["rocprofiler-sdk-tool"]["strings"][
+                        "buffer_records"
+                    ]:
+                        if item["kind"] == "MARKER_CORE_RANGE_API":
+                            string_records = item["operations"]
+                    exclude_ops = {"roctxMarkA", "roctxGetThreadId"}
```

Comment:
```suggestion
                    exclude_ops = {"roctxGetThreadId"}
```
You can also probably remove "roctxMarkA" from the exclude list.  According to generateRocpd.cpp, the timestamps comparison should take care of moving those entries to samples instead of regions.  We don't need to purposely exclude any roctxMarkA operations.

Reply:
roctxMarkA removed.